### PR TITLE
refactor: add `resource_reader` module

### LINKED ISSUE #1618: refactor: Create a Resource Reader
Create a module `resource_reader.rs` that reads a file from local or absolute paths or an HTTP URL. This implementation currently resides in the `config::reader` module. Refactor it and take it out into the root module. The `FileRead` type which is defined in the `config::reader` should also be moved out into `resource_reader.rs`.

**Technical Requirements**
- Ensure the tests are migrated from config::reader

### PR DESCRIPTION
**Pull request by webbdays**
**Summary:**  
moved resource read logic into a new module in root.

**Issue Reference(s):**  
Fixes #1618

/claim #1618

**Build & Testing:**

- [Yes] I ran `cargo test` successfully.
- [Yes] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [Yes ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **Refactor**
	- Improved modularity and efficiency in configuration file reading operations by introducing a dedicated `ResourceReader`.

### CODE REVIEW

File: src/config/reader.rs [hunk 1]
```diff
@@ -24,57 +22,15 @@ use crate::valid::{Valid, Validator};
 /// linked extensions to create a ConfigModule.
 pub struct ConfigReader {
     runtime: TargetRuntime,
```
Comment (ssddOnTop) [inline src/config/reader.rs:24]: do we need to store runtime in config reader anymore?

Comment (webbdays) [reply src/config/reader.rs:24]: Hi @ssddOnTop,
thanks for the review.
https://github.com/tailcallhq/tailcall/blob/57ca2e423193899ac4582b5b6663844f151f02bc/src/config/reader.rs#L219
it being used here.
Any other way to get it?

**Review by ssddOnTop — DISMISSED** [review]
`load_cert` and `load_private_key` should be moved as well

**Comment by webbdays** [comment]
Ok.

**Comment by webbdays** [comment]
@ssddOnTop

I think for those, content is already read right.
They just use already read content using read_file.
I think they should not be included in resource reader file.
Thank you.

File: src/config/reader.rs (hunk 1 above)
Comment (tusharmath) [reply src/config/reader.rs:24]: Let's keep it for now.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**